pkg/provisioner/ironic/power_test.go

```diff
@@ -199,7 +199,8 @@ func TestPowerOff(t *testing.T) {
 			}
 
 			prov.status.ID = nodeUUID
-			result, err := prov.PowerOff()
+			// We pass false here to default to a soft power off
+			result, err := prov.PowerOff(false)
```

Comment:
I'd add a new boolean variable to the test-case struct above, then this can be e.g `tc.hardMode` - and we can add some new test-cases that exercise the new code path with `hardMode=true`

Reply:
Yep, good idea. I've got that covered.